src/cargo/core/source.rs

```diff
@@ -27,7 +27,7 @@ pub trait Source: Registry {
 
     /// The download method fetches the full package for each name and
     /// version specified.
-    fn download(&mut self, package: &PackageId) -> CargoResult<Package>;
+    fn download(&mut self, package: &PackageId, config: &Config) -> CargoResult<Package>;
```

Comment:
Most of these `Source` types already store `Config`, so I think it may not need to get passed in here?